frontend/src/components/GoesData/TagModal.tsx

```
@@ -53,9 +53,14 @@ export default function TagModal({ frameIds, onClose }: Readonly<{ frameIds: str
   };
 
   return (
-    <div className="fixed inset-0 bg-black/30 dark:bg-black/60 flex items-center justify-center z-50 modal-overlay" onClick={onClose}>
-      <div ref={dialogRef} role="dialog" aria-label="Tag Frames" aria-modal="true"
-        className="bg-gray-50 dark:bg-slate-900 rounded-xl p-6 border border-gray-200 dark:border-slate-700 w-96 space-y-4 modal-panel" onClick={(e) => e.stopPropagation()}>
+    <dialog
+      open
+      className="fixed inset-0 bg-black/30 dark:bg-black/60 flex items-center justify-center z-50 modal-overlay m-0 w-full h-full max-w-none max-h-none border-none"
+      onClick={(e) => { if (e.target === e.currentTarget) onClose(); }}
+    >
```

Biome (2.3.14)

[error] Enforce to have the onClick mouse event with the onKeyUp, the onKeyDown, or the onKeyPress keyboard event.

Actions triggered using mouse events should have corresponding keyboard events to account for keyboard-only navigation.

(lint/a11y/useKeyWithClickEvents)